db/background_mgr.go

```diff
@@ -52,53 +85,181 @@ type BackgroundManagerStatus struct {
 // BackgroundManagerProcessI is an interface satisfied by any of the background processes
 // Examples of this: ReSync, Compaction
 type BackgroundManagerProcessI interface {
-	Run(options map[string]interface{}, terminator chan struct{}) error
+	Init(options map[string]interface{}, clusterStatus []byte) error
+	Run(options map[string]interface{}, persistClusterStatusCallback updateStatusClusterAwareCallback, terminator chan struct{}) error
 	GetProcessStatus(status BackgroundManagerStatus) ([]byte, error)
 	ResetStatus()
 }
 
+type updateStatusClusterAwareCallback func() error
+
 func (b *BackgroundManager) Start(options map[string]interface{}) error {
 	err := b.markStart()
 	if err != nil {
 		return err
 	}
 
+	var processClusterStatus []byte
+	if b.clusterAwareOptions != nil {
```

Comment:
We're checking if `b.clusterAwareOptions != nil' in a lot of places.  It wasn't immediately obvious to me that this is equivalent to checking whether the background manager is cluster aware or is just local-only.  

I suppose it's a nitpick, but either a b.isClusterAware() function that was is checking the Options value under the hood, or a b.isClusterAware bool on the struct would make this more clear.